sklearn/metrics/regression.py

```diff
@@ -573,3 +574,46 @@ def r2_score(y_true, y_pred, sample_weight=None,
         avg_weights = multioutput
 
     return np.average(output_scores, weights=avg_weights)
+
+
+def max_error(y_true, y_pred, sample_weight=None):
```

Comment:
I'm not convinced we should support `sample_weight`. (Do you have good basis in the literature to do so?) It doesn't have the same meaning here as elsewhere. Elsewhere `metric(np.repeat(y_true, weight), np.repeat(y_pred, weight)) == metric(y_true, y_pred, sample_weight=weight)` holds for integer array weight. Here it will not.